tests/telemetry/test_events.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import pytest
+import json
+import os
+import sys
+import time
+
+from telemetry_utils import skip_201911_and_older
+
+pytestmark = [
+    pytest.mark.topology('any')
+]
+
+EVENTS_TESTS_PATH = "./telemetry/events"
+sys.path.append(EVENTS_TESTS_PATH)
+
+
+logger = logging.getLogger(__name__)
+
+BASE_DIR = "logs/telemetry"
+DATA_DIR = os.path.join(BASE_DIR, "files")
+
+CMD_PREFIX_FMT = ("{} -client_types=gnmi -a {}:50051 -t EVENTS "
+                  "-logtostderr -insecure -v 7 -streaming_type ON_CHANGE "
+                  "-qt s -q all")
+CMD_PREFIX = ""
+
+GNMI_CLI_BIN = None
+
+
+def validate_yang(duthost, op_file="", yang_file=""):
+    assert op_file != "" and yang_file != "", "op_file path or yang_file name not provided"
+    cmd = "python /tmp/validate_yang_events.py -f {} -y {}".format(op_file, yang_file)
+    logger.info("Performing yang validation on {} for {}".format(op_file, yang_file))
+    ret = duthost.shell(cmd)
+    assert ret["rc"] == 0, "Yang validation failed for {}".format(yang_file)
+
+
+def run_cmd(localhost, params={}, op_file="", filter_event="", event_cnt=0, timeout=0):
+    cmd = CMD_PREFIX
+    for i in params:
+        cmd += "[{}]".format(i)
+
+    if (op_file != ""):
+        cmd += " -output_file={}".format(op_file)
+
+    if (filter_event != ""):
+        cmd += " -expected_event={}".format(filter_event)
+
+    if (event_cnt > 0):
+        cmd += " -expected_count={}".format(event_cnt)
+
+    if (timeout > 0):
+        cmd += " -streaming_timeout={}".format(timeout)
+
+    ret = localhost.shell(cmd)
+    assert ret["rc"] == 0, "Failed to run cmd {}".format(cmd)
+
+
+def do_init(duthost):
+    global CMD_PREFIX, GNMI_CLI_BIN
+
+    for i in [BASE_DIR, DATA_DIR]:
+        try:
+            os.mkdir(i)
+        except OSError as e:
+            logger.info("Dir/file already exists: {}, skipping mkdir".format(e))
+
+    duthost.shell("docker cp telemetry:/usr/sbin/gnmi_cli /tmp")
+    ret = duthost.fetch(src="/tmp/gnmi_cli", dest=DATA_DIR)
+    GNMI_CLI_BIN = ret.get("dest", None)
+    assert GNMI_CLI_BIN is not None, "Failing to get gnmi_cli"
```

Comment:
test_telemetry.py is using gnmi_cli in ptf container, and test_events.py is using gnmi_cli in sonic_mgmt container, can you confirm?

Reply:
Yes

Reply:
If DUT is using arm platform, you can't copy gnmi_cli from dut.

Reply:
Skipping arm platform